submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/service/filesystem/FilesService.kt

```diff
@@ -122,25 +124,32 @@ class FilesService(
 
     private fun copy(
         extFile: ExtFile,
-        folder: File,
+        tempFolder: File,
+        submissionFilesFolder: File,
         filePermissions: Set<PosixFilePermission>,
         folderPermissions: Set<PosixFilePermission>
     ) {
         val source = extFile.file
-        val target = folder.resolve(extFile.fileName)
-
-        logger.info { "copying file ${source.absolutePath} into ${target.absolutePath}" }
-        copyOrReplaceFile(source, target, filePermissions, folderPermissions)
+        val target = tempFolder.resolve(extFile.fileName)
+        val current = submissionFilesFolder.resolve(extFile.fileName)
+
+        if (current.notExist() || source.md5() != current.md5()) {
```

Comment:
not the following the logic here, why if md5 has not been changed we copy the file while if md5 is different we move it?